pkg/converter/converter.go

```diff
@@ -0,0 +1,486 @@
+	}
+
+	return result
+}
+
+func convertConfigs(context *ConfigConvertContext, environments map[string]manifest.EnvironmentDefinition,
+	configs []configv1.Config) (projectv2.ConfigsPerApisPerEnvironments, []error) {
+
+	result := make(projectv2.ConfigsPerApisPerEnvironments)
+	var errors []error
+
+	for _, conf := range configs {
+		for _, env := range environments {
+			if _, found := result[env.Name]; !found {
+				result[env.Name] = make(map[string][]configv2.Config)
+			}
+
+			convertedConf, configErrors := convertConfig(context, env, conf)
+
+			if configErrors != nil {
+				errors = append(errors, configErrors...)
+				continue
+			}
+
+			result[env.Name][conf.GetApi().GetId()] = append(result[env.Name][conf.GetApi().GetId()], convertedConf)
+		}
+	}
+
+	if errors != nil {
+		return nil, errors
+	}
+
+	return result, nil
+}
+
+func convertConfig(context *ConfigConvertContext, environment manifest.EnvironmentDefinition, config configv1.Config) (configv2.Config, []error) {
+	var errors []error
+
+	coord := coordinate.Coordinate{
+		Project: context.ProjectId,
+		Api:     config.GetApi().GetId(),
+		Config:  config.GetId(),
+	}
+
+	templ, envParams, errs := convertTemplate(context, coord, config.GetFilePath())
+
+	if len(errs) > 0 {
+		errors = append(errors, &ConvertConfigError{
+			Location: coord,
+			Reason:   fmt.Sprintf("unable to load template `%s`: %s", config.GetFilePath(), errs),
+		})
+	}
+
+	parameters, references, skip, parameterErrors := convertParameters(context, environment, config)
+
+	if parameterErrors != nil {
+		errors = append(errors, parameterErrors...)
+	}
+
+	for paramName, param := range envParams {
+		if _, found := parameters[paramName]; found {
+			errors = append(errors, &ConvertConfigError{
+				Location: coord,
+				Reason: fmt.Sprintf("parameter name collision. automatic environment variable conversion failed. please rename `%s` parameter",
+					paramName),
+			})
+			continue
+		}
+
+		parameters[paramName] = param
+	}
+
+	if errors != nil {
+		return configv2.Config{}, errors
+	}
+
+	return configv2.Config{
+		Template:    templ,
+		Coordinate:  coord,
+		Group:       environment.Group,
+		Environment: environment.Name,
+		Parameters:  parameters,
+		References:  references,
+		Skip:        skip,
+	}, nil
+}
+
+func convertTemplate(context *ConfigConvertContext, coordinates coordinate.Coordinate,
+	templatePath string) (template.Template, map[string]parameter.Parameter, []error) {
+	data, err := afero.ReadFile(context.Fs, templatePath)
+
+	if err != nil {
+		return nil, nil, []error{err}
+	}
+
+	environmentParameters := map[string]parameter.Parameter{}
+	var errors []error
+
+	templText := envVariableRegex.ReplaceAllStringFunc(string(data), func(p string) string {
+		match := envVariableRegex.FindStringSubmatch(p)
+
+		if len(match) != 2 {
+			errors = append(errors, fmt.Errorf("cannot parse environment variable: `%s` seems to be invalid", p))
+			return ""
+		}
+
+		envVar := string(match[1])
```

Comment:
No need for type conversion
```suggestion
		envVar := match[1]
```